Common/cpp/ReanimatedRuntime/WorkletRuntime.cpp

```diff
@@ -4,15 +4,68 @@
 #include "WorkletRuntimeCollector.h"
 #include "WorkletRuntimeDecorator.h"
 
+#include <jsi/decorator.h>
+
 namespace reanimated {
 
+class AroundLock {
+  const std::shared_ptr<std::recursive_mutex> mutex_;
+
+ public:
+  explicit AroundLock(const std::shared_ptr<std::recursive_mutex> &mutex)
+      : mutex_(mutex) {}
+
+  void before() const {
+    mutex_->lock();
+  }
+  void after() const {
+    mutex_->unlock();
+  }
+};
+
+class LockableRuntime : public jsi::WithRuntimeDecorator<AroundLock> {
+  AroundLock aroundLock_;
+  std::shared_ptr<jsi::Runtime> runtime_;
```

Comment:
Why do we need to store `runtime_`? Where do we use it?

Reply:
It seems that this is required by jsi::RuntimeDecorator. If I remove it the application will crash :/